File: app/src/main/java/com/guideapp/data/remote/GuideApi.java
package com.guideapp.data.remote;

import com.guideapp.model.Local;
import com.guideapp.model.SubCategory;
import com.guideapp.model.wrapper.ListResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface GuideApi {

    @GET("local")
    Call<ListResponse<Local>> getLocals(
            @Query("idCity") Long idCity,
            @Query("idCategory") Long idCategory,
            @Query("idSubCategories") long[] idSubCategory);

    @GET("local")
    Call<ListResponse<Local>> getLocals(
            @Query("idCity") Long idCity);

    @GET("subcategory")
    Call<ListResponse<SubCategory>> getSubCategories(@Query("idCategory") long idCategory);
}

File: app/src/main/java/com/guideapp/data/remote/GuideApi.kt
package com.guideapp.data.remote

import com.guideapp.model.Local
import com.guideapp.model.SubCategory
import com.guideapp.model.wrapper.ListResponse

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface GuideApi {

    @GET("local")
    fun getLocals(
            @Query("idCity") idCity: Long?,
            @Query("idCategory") idCategory: Long?,
            @Query("idSubCategories") idSubCategory: LongArray): Call<ListResponse<Local>>

    @GET("local")
    fun getLocals(
            @Query("idCity") idCity: Long?): Call<ListResponse<Local>>

    @GET("subcategory")
    fun getSubCategories(@Query("idCategory") idCategory: Long): Call<ListResponse<SubCategory>>
}
